boards/itsybitsy_m4/examples/sercom_interrupt.rs

```diff
@@ -0,0 +1,160 @@
+#![no_std]
+#![no_main]
+
+//! The idea of this example is to show two different things
+//! - How to manually configure SERCOM for a desired mode (UART in this case)
+//! - How to enable an use SERCOM interrupts
+//! The second one is particularly handy because some drivers will give
+//! you a `poll()` or `update()` function, and you are expected to call it  
+//! on the receive interrupt of the peripheral i.e. SERCOM UART RXC
+//! (receive complete flag).
+//!
+//! This example is basically just a blinky script. The MCU sends itself
+//! a byte via UART that will trigger the RXC interrupt and then toggles
+//! the builtin red led
+//!
+//! Wiring
+//! - RX - A4
+//! - TX - A5
+//!
+//! > In the ATSAMX5x MCUs SERCOM UART configuration requires at least
+//! two pins from the same IOSET. You can find more info about this in
+//! the hal documentation or in the MCU datasheet at:
+//! <https://www.microchip.com/en-us/product/ATSAMD51G19A>
+
+use itsybitsy_m4 as bsp;
+#[cfg(not(feature = "use_semihosting"))]
+use panic_halt as _;
+#[cfg(feature = "use_semihosting")]
+use panic_semihosting as _;
+
+use bsp::entry;
+use core::cell::RefCell;
+use cortex_m::{interrupt::Mutex, peripheral::NVIC};
+
+use bsp::hal::{
+    clock::GenericClockController,
+    delay::Delay,
+    ehal::blocking::delay::DelayMs,
+    gpio::v2::{AlternateD, Pin, PushPullOutput, PA04, PA06, PA22},
+    pac::{self, interrupt, CorePeripherals, Peripherals},
+    prelude::*,
+    sercom::v2::{
+        uart::{self, BaudMode, Flags, Oversampling},
+        IoSet3, Sercom0,
+    },
+    time::Hertz,
+};
+
+/// Pin labeled as A5
+type UartRx0 = Pin<PA06, AlternateD>;
+/// Pin labeled as A4
+type UartTx0 = Pin<PA04, AlternateD>;
```

Comment:
Both this and `UartRx0` could benefit from being aliases in the `bsp_pins!` declaration instead of only local to this example

Reply:
These pins are not by default UART pins, that's why I declare them in the example. 
The BSPs are supposed to alias these pins only for the labeled functions IIRC, but these pins correspond to a sercom that is left available for the user

Reply:
Aliases are exactly for this purpose. If there is an alternate function you'd potentially like to use you can list it as an alias and the macro automatically makes these type declarations. The name of the pin should be it's typical use case

Reply:
Ok the thing is, that's not the typical use case (?
I can add them tho', but about the naming convention, wouldn't it be better then to do as suggested by @bradleyharden  on the matrix channel? To give the pins alias based on the SercomPad and not the specific function: 

```Rust
    PA16 {
        name: d0_rx
        aliases: {
            AlternateD: UartRx,
            AlternateD: Sercom3Pad1,
        }
    }
    PA17 {
        name: d1_tx
        aliases: {
            AlternateD: UartTx,
            AlternateD: Sercom3Pad0,
        }
    }
```

Reply:
Thinking about it it would also be nice to have these aliases directly given by the hal as they would be board-agnostic

Reply:
I'm neutral on the exact name, I just think it's useful to be in the HAL for the reason you just gave

Reply:
This hasn't been fixed yet

Reply:
Hmm, now that I think about this more, `Sercom3Pad0` is not unique. There are several pins that could apply to. Maybe that's not as good a name as I thought it was.

Reply:
Ah, I see you added the `IoSet` to the name. That makes it unique. But that wouldn't work for `thumbv6`. I wouldn't mind adding some aliases to the HAL. They could probably be auto-generated from the existing macro calls. I just want to make sure they are unique.

Reply:
For the Xiao case I went with the pin name precisely because there are no Iosets. An alternative would be to name groups of pads as iosets but add to the documentation that this is only an abstraction we found useful, like what zephyr does with the esp32 - divide gpio in 2 different ports not because that's how they are divided irl but because it is useful to have them divided that way